homeassistant/components/nam/air_quality.py

```diff
@@ -51,23 +71,21 @@ def __init__(self, coordinator: NAMDataUpdateCoordinator, sensor_type: str) -> N
     @property
     def particulate_matter_2_5(self) -> StateType:
         """Return the particulate matter 2.5 level."""
-        return round_state(
-            getattr(self.coordinator.data, f"{self.sensor_type}{SUFFIX_P2}")
+        return cast(
+            StateType, getattr(self.coordinator.data, f"{self.sensor_type}{SUFFIX_P2}")
```

Comment:
particulate_matter_2_5, actually cannot be `StateType`. StateType includes types like a string, which is not correct; `particulate_matter_2_5` can be a float or None.

Reply:
Done. One question, library passes `dataclass` object with type annotations to HA but this type annotation is missing via `coordinator.data`. There is solution for this?

Reply:
This is caused by accessing it dynamically via `getattr`, a static type checker can never know what the result will be.

Reply:
But If I access sensor value like this `coordinator.data.mhz14a_carbon_dioxide`, `mypy` also thinks the type is `Any` (in fact it's `int`). Maybe something is missing in my code?